src/ddprof_worker.cc

```diff
@@ -317,76 +329,73 @@ void ddprof_pr_lost(DDProfContext *, perf_event_lost *lost, int) {
   ddprof_stats_add(STATS_EVENT_LOST, lost->lost, NULL);
 }
 
-void ddprof_pr_comm(DDProfContext *ctx, perf_event_comm *comm, int pos) {
+void ddprof_pr_comm(DDProfContext *ctx, perf_event_comm *comm,
+                    int watcher_pos) {
   // Change in process name (assuming exec) : clear all associated dso
   if (comm->header.misc & PERF_RECORD_MISC_COMM_EXEC) {
-    LG_DBG("<%d>(COMM)%d -> %s", pos, comm->pid, comm->comm);
+    LG_DBG("<%d>(COMM)%d -> %s", watcher_pos, comm->pid, comm->comm);
     unwind_pid_free(ctx->worker_ctx.us, comm->pid);
   }
 }
 
-void ddprof_pr_fork(DDProfContext *ctx, perf_event_fork *frk, int pos) {
-  LG_DBG("<%d>(FORK)%d -> %d/%d", pos, frk->ppid, frk->pid, frk->tid);
+void ddprof_pr_fork(DDProfContext *ctx, perf_event_fork *frk, int watcher_pos) {
+  LG_DBG("<%d>(FORK)%d -> %d/%d", watcher_pos, frk->ppid, frk->pid, frk->tid);
   if (frk->ppid != frk->pid) {
     // Clear everything and populate at next error or with coming samples
     unwind_pid_free(ctx->worker_ctx.us, frk->pid);
   }
 }
 
-void ddprof_pr_exit(DDProfContext *ctx, perf_event_exit *ext, int pos) {
+void ddprof_pr_exit(DDProfContext *ctx, perf_event_exit *ext, int watcher_pos) {
   // On Linux, it seems that the thread group leader is the one whose task ID
   // matches the process ID of the group.  Moreover, it seems that it is the
   // overwhelming convention that this thread is closed after the other threads
   // (upheld by both pthreads and runtimes).
   // We do not clear the PID at this time because we currently cleanup anyway.
   (void)ctx;
   if (ext->pid == ext->tid) {
-    LG_DBG("<%d>(EXIT)%d", pos, ext->pid);
+    LG_DBG("<%d>(EXIT)%d", watcher_pos, ext->pid);
   } else {
-    LG_DBG("<%d>(EXIT)%d/%d", pos, ext->pid, ext->tid);
+    LG_DBG("<%d>(EXIT)%d/%d", watcher_pos, ext->pid, ext->tid);
   }
 }
 
 /********************************** callbacks *********************************/
-DDRes ddprof_worker_maybe_export(DDProfContext *ctx, int64_t now_ns,
-                                 bool *restart_worker) {
```

Comment:
Not sure about this, having `restart_worker` as output parameter makes it clear that this function determine if worker should be restarted or not.